src/cider/nrepl/middleware/stacktrace.clj

```diff
@@ -31,20 +34,47 @@
   [frame flag]
   (update-in frame [:flags] (comp set conj) flag))
 
+(defn- source-path
+  "Return the relative source path for the top-level class without extension."
+  [class]
+  (-> (str/replace (str class) #"\$.*" "")
+      (str/replace "." "/")))
+
 (defn analyze-fn
   "Add namespace, fn, and var to the frame map when the source is a Clojure
   function."
   [{:keys [file type class method] :as frame}]
-  (if (= :clj type)
+  (if (or (= :clj type)
+          (= :cljc type))
     (let [[ns fn & anons] (-> (repl/demunge class)
                               (str/replace #"--\d+" "")
                               (str/split #"/"))
           fn (or fn method)] ; protocol functions are not munged
       (assoc frame
              :ns  ns
              :fn  (str/join "/" (cons fn anons))
-             :var (str ns "/" fn)))
-    frame))
+             :var (str ns "/" fn)
+             ;; Full file path
+             :full-file (or (some-> (info/info-clj 'user (symbol (str ns "/" fn)))
+                                    :file
+                                    info/file-info
+                                    :file
+                                    u/transform-value)
+                            ;; invoke/invokeStatic
```

Comment:
I don't quite get why this second code branch is needed to be honest.

Reply:
I can improve the comment, but I noticed that invoke & invokeStatic don't get their var extracted from the frame correctly. So this uses a different approach to grab the value of the file.

Reply:
Got it. I wonder if that's some Clojure bug, that should be reported upstream as well.

Reply:
Okay, I've dug in a little more, it actually serves an unexpected purpose purely during reload. As the ns & var is unloaded by refresh, getting a source location is not possible. However, you can figure out the file that should hold it, and that will be on the classpath.

invokeStatic was a distraction I accidentally bumped into, as I was doing lightweight tests with only invoke/invokeStatic in the frame. I now suspect all clojure frames are like that!

I'll update the comment accordingly.

Reply:
Ah, it also handles a second case! Top-level evaluations have a class like `aero.core_foo$eval16976` or `aero.core-foo/eval16976` as a clojure var. This doesn't have any metadata though, as it isn't "real".

This code path is able to infer the correct filepath based on this fake var.

The reason to not always use this code path is in the case that you load in a function dynamically, although I wouldn't expect this to be commonplace.